rest-service/manager_rest/storage/resource_models.py

```diff
@@ -532,4 +534,36 @@ def set_node(self, node):
         self._set_parent(node)
         self.node = node
 
+
+class Agent(CreatedAtMixin, SQLResourceBase):
+    __tablename__ = 'agents'
+
+    ip = db.Column(db.Text)
+    name = db.Column(db.Text)
+    install_method = db.Column(db.Text)
+    system = db.Column(db.Text)
+    version = db.Column(db.Text)
+    state = db.Column(db.Enum(*AgentState.STATES, name='agent_states'),
+                      default=AgentState.CREATING)
+    rabbitmq_username = db.Column(db.Text)
+    rabbitmq_password = db.Column(db.Text)
+    rabbitmq_exchange = db.Column(db.Text)
+    updated_at = db.Column(UTCDateTime)
+
+    _node_instance_fk = foreign_key(NodeInstance._storage_id)
+
+    @declared_attr
+    def node_instance(cls):
+        return one_to_many_relationship(cls,
```

Comment:
Explain why this is one to many

Reply:
proxy agents i guess

Reply:
I think it was more for upgraded agents, where you'd have 2 connected to the same instance.

Reply:
Yes, I added a comment